pd/src/pending_block.rs

```diff
@@ -16,14 +17,21 @@ pub struct PendingBlock {
     pub note_commitment_tree: NoteCommitmentTree,
     /// Stores note commitments for convienience when updating the NCT.
     pub notes: BTreeMap<note::Commitment, PositionedNoteData>,
+    /// Notes that are undelegation outputs in this block (not to be committed to the NCT yet).
+    pub undelegation_notes: BTreeMap<note::Commitment, QuarantinedNoteData>,
```

Comment:
Should this be called `quarantined_notes` to match `quarantined_nullifiers`?

Reply:
I ended up using a slightly different architecture, which avoids duplicating data by grouping nullifiers and notes by transaction, obviating this (valid) suggeestion.